test/utils/document.js

```diff
@@ -0,0 +1,41 @@
+/* global window */
+
+export default class DocumentTest {
+  // Simplify the testing of deck.gl work that requires a document and canvas
+  constructor(canvasCallback) {
+    this.document = this._createDocument();
+    this.canvasParent = null;
+    this.testObject = null;
+    this._setup(canvasCallback);
+  }
+
+  _createDocument() {
+    // Creates a global document object via JSDOM or a deep clone of the browser one if available
+    if (typeof window === undefined || !window.document) {
+      const {JSDOM} = require('jsdom');
+      const dom = new JSDOM(`<!DOCTYPE html>`);
+      return dom.window.document;
+    }
+    const documentClone = window.document.cloneNode(true);
```

Comment:
Why do you need to clone the document? It may have heavy content during browser test.